### PR TITLE
Refactor session 1 - Stat tracker methods and parser arguments

### PR DESCRIPTION
**Pull request by aevans27**
## Description
Stat tracker methods and parser arguments

Please include a description of what was changed
Update to the stat tracker class to call methods required for the harness
Change to parsers to take arguments for paths rather than hard coding it
Change to requires to relatives so harness can access project and classes
## Type of change

- [ ] fix
- [x] feat
- [x] test
- [x] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [ ] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

File: lib/game_parser.rb [hunk 1]
```diff
@@ -2,15 +2,15 @@
 # We will use game_id, season, away_team_id, hone_team_id, away_goals, home_goals
 
 require "csv"
-require './lib/game'
+require_relative './game'
```
Comment (ttakahashi1591) [inline lib/game_parser.rb:5]: Glad this finally worked out!

File: lib/team_result_parser.rb [hunk 2]
```diff
@@ -130,19 +135,22 @@ def alltime_goals_per_team #helper method
     goals_hash
   end
   
-  def best_offense(teams)
+  def best_offense
     best_offense = nil
+    @league_parser.list_teams(@team_path)
+    teams = @league_parser.get_team_list
```
Comment (ttakahashi1591) [inline lib/team_result_parser.rb:141]: This was a great idea in order to ensure that we did not need an argument for this method!

**Review by ttakahashi1591 — APPROVED** [review]
Really great job with going through and refactoring. Super exciting to to see that our methods are passing the `spec_harness` test as well!!!

File: lib/team_result_parser.rb [hunk 3]
```diff
@@ -1,13 +1,16 @@
 require "csv"
-require './lib/team_result'
+require_relative './team_result'
+require_relative './league_parser'
 class TeamResultParser
   attr_reader :game_data_by_team
   def initialize
     @game_data_by_team = []
+    @league_parser = LeagueParser.new
```
Comment (lanebret85) [inline lib/team_result_parser.rb:8]: brilliant move instantiating @league_parser here to be able to call on the teams.csv data without passing in an argument!

File: spec/stat_tracker_spec.rb
Comment (lanebret85) [inline spec/stat_tracker_spec.rb:1]: stat tracker tests look great for all methods we have completed as of now!

File: lib/team_result_parser.rb (hunk 3 above)
Comment (TLamb32) [inline lib/team_result_parser.rb:8]: Now that you explained this to me, it makes so much sense.

**Review by TLamb32 — APPROVED** [review]
Great work, the stat tracker looks so clean.

**Review by lanebret85 — APPROVED** [review]

**Pull request merged**